On The Map/Controller/AddLocationCompletionViewController.swift

```diff
@@ -0,0 +1,64 @@
+//
+//  AddLocationCompletionViewController.swift
+//  On The Map
+//
+//  Created by Michal Hus on 2/23/20.
+//  Copyright © 2020 Michal Hus. All rights reserved.
+//
+
+import UIKit
+import MapKit
+import CoreLocation
+
+class AddLocationCompletionViewController: UIViewController, MKMapViewDelegate {
+   
+    var coordinate: CLLocationCoordinate2D = CLLocationCoordinate2D()
+    var link:String = ""
+    var mapString: String = ""
```

Comment:
Two small things to think about with these:

The link and the mapString are required data for this view controller to exist; required components are referred to as "dependencies" and the way you pass an object the components it needs to do its job is called "dependency injection". Typically we do that in the initializer of an object (pass in what it needs as params to the initializer) but for viewControllers that are initialized via storyboard, we don't have that option, so things get a little weird and we have some choices about how to handle it.

First, your way of handling it here (use non-optionals with default values) is pretty good so don't sweat it too much!

If you have more than one object that needs to be injected into a class somehow (i.e. in this case, where you assign them in `prepareForSegue`) and you can't pass them to an initializer, you can make it more explicit by adding a `func inject(link: String, mapString: String)`. Sometimes we do that because 1) code that's more explicit is easier to understand and 2) assigning each property one by one will get ugly if there are more than two of them. Again, not a big deal.

Secondly, the whole "view controllers initialized from storyboards" is one case where I would argue implicitly unwrapped optionals might be preferable to non-optionals with invalid default values. For example:
`var link: String!` and `var mapString: String!` instead of initializing them with `""`. Part of the reason for this is that `""` is a reasonable default value for a string, but it's not really a valid value for these properties of this view controller. Neither is perfect but worth considering.

Reply:
Let's talk about this a little more viewControllers that are initialized via a storyboard.